sailthru/sailthru_content/services/catalog_api_service.py

```diff
@@ -44,10 +45,43 @@ def _get_resource_from_api(self, api_endpoint, page_size, **kwargs):
 
         return results
 
+    def _get_resource_from_search_api(self, api_endpoint, **kwargs):
+        page = 1
+        results = []
+        next_page = True
+        while next_page:
+            response = api_endpoint.get(**kwargs)
+            search_response = response.get('objects')
+            results += search_response.get('results', [])
+            next_page = search_response.get('next')
+            page += 1
+            kwargs['page'] = page
+
+        return results
+
     def get_courses(self):
         logger.debug('Get Courses called')
         return self._get_resource_from_api(self.api_client.courses(), COURSES_PAGE_SIZE, marketable=1)
 
+    def get_searchable_course_keys(self):
+        logger.debug('Get Searchable Courses called')
+        searchable_course_keys = []
+        searchable_courses = self._get_resource_from_search_api(self.api_client.search.all.facets())
```

Comment:
I think you should test it.  In fact, I don't think you need to provide the marketable query string parameter.  If we look at the marketing site code:

https://github.com/edx/edx-mktg/blob/master/docroot/sites/all/modules/features/edx_catalog_api_proxy/edx_catalog_api_proxy.module#L362

We only provide the search query and also pass in the content type (courserun, program) in the query string.  Can you model your approach on this at least and pass in the content type, or even better, find out if the course run search endpoint will give you what you need?